llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -2776,6 +2776,12 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
       }
       return Cost;
     }
+    case Intrinsic::maximumnum:
+    case Intrinsic::minimumnum: {
+      if (TLI->isOperationLegalOrPromote(llvm::ISD::FMAXNUM_IEEE, LT.second))
```

Comment:
Don't need llvm::

Technically you should also check FMINNUM_IEEE for minimumnum, it's just unlikely these ever differ in practice

Reply:
If so, the problem will be quite complicated: different architecture uses different `fcanonical`.
LoongArch uses `min`
MIPSr6 uses `max`
AArch64 uses `max`
Maybe we can use `max && min` here.

Reply:
I don't think this really needs to be that detailed. Plus you can always directly query for the cost of the canonicalize intrinsic (some of the other costs here have long sequences of getIntrinsicInstrCost with some assumed expansion, but it's kind of annoying).

Should check the right ISD for the right intrinsic too